src/Microsoft.OData.Core/UriParser/Resolver/ODataUriResolver.cs

```diff
@@ -91,6 +91,27 @@ public virtual IEdmNavigationSource ResolveNavigationSource(IEdmModel model, str
                 return navSource;
             }
 
+            if (model.IsImmutable())
+            {
+                NormalizedModelElementsCache cache = GetNormalizedModelElementsCache(model);
+                IList<IEdmNavigationSource> cachedResults = cache.FindNavigationSources(identifier);
+
+                if (cachedResults != null)
+                {
+                    if (cachedResults.Count == 1)
+                    {
+                        return cachedResults[0];
+                    }
+
+                    if (cachedResults.Count > 1)
+                    {
+                        throw new ODataException(Strings.UriParserMetadata_MultipleMatchingNavigationSourcesFound(identifier));

```

Comment:
why throw an exception is cachedResults are more than one and you said this is a possibility? or what does this mean?

Reply:
This cache-based code path replicates the existing behaviour of the `ResolveNavigationSource` method, which is to throw an exception if multiple matches are found.

Notice that this method returns a single `IEdmNavigationSource`, not a collection. If there are multiple matches, then it means the URI resolver doesn't know which one to use and throws an exception.